src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/CompareUtil.java

```diff
@@ -52,20 +65,42 @@
      */
     char getOperator();
 
+    abstract class AbstractCompareUtil implements CompareUtil {
+        @Override
+        public boolean matches(String pattern, String str) {
+            return matches(pattern, str, null);
+        }
+
+        /**
+         * Expands the pattern in case it contains tokens that can be resolved with the given envVars.
+         * @param pattern the pattern to expand.
+         * @param envVars the envVars to use for expansion, if {@code null} then the unmodified pattern will be returned
+         * @return the expanded string, if no envVars were provided will return the initial unmodified pattern
+         */
+        protected String expandWithEnvVarsIfPossible(String pattern, EnvVars envVars) {
+            if (envVars != null) {
+                return envVars.expand(pattern);
+            } else {
+                return pattern;
+            }
+        }
+    }
+
     /**
      * Compares based on Ant-style paths.
      * like <code>my/&#042;&#042;/something&#042;.git</code>
      */
-    static class AntCompareUtil implements CompareUtil {
+    class AntCompareUtil extends AbstractCompareUtil {
 
         @Override
-        public boolean matches(String pattern, String str) {
+        public boolean matches(String pattern, String str, EnvVars envVars) {
             // Replace the Git directory separator character (always '/')
             // with the platform specific directory separator before
             // invoking Ant's platform specific path matching.
             String safePattern = pattern.replace('/', File.separatorChar);
+            String expandedPattern = expandWithEnvVarsIfPossible(safePattern, envVars);
```

Comment:
I think it is better to do this before the `\` to `/` replacement in case any of the used env vars contains platform specific paths.